specs/deneb/polynomial-commitments.md

```diff
@@ -447,72 +465,92 @@ def compute_kzg_proof_impl(polynomial: Polynomial, z: BLSFieldElement) -> KZGPro
-    and (3) the polynomial evaluation random challenge.
-    This function should also work with blobs == [] and kzg_commitments == []
+    Given a blob, return the KZG proof that is used to verify it against the commitment.
+    Public method.
     """
-    assert len(blobs) == len(kzg_commitments)
+    commitment = blob_to_kzg_commitment(blob)
+    polynomial = blob_to_polynomial(blob)
+    evaluation_challenge = compute_challenge(blob, commitment)
+    return compute_kzg_proof_impl(polynomial, evaluation_challenge)
+```
 
-    # Convert blobs to polynomials
-    polynomials = [blob_to_polynomial(blob) for blob in blobs]
+#### `verify_blob_kzg_proof`
 
-    # Generate random linear combination and evaluation challenges
-    r_powers, evaluation_challenge = compute_challenges(polynomials, kzg_commitments)
+```python
+def verify_blob_kzg_proof(blob: Blob,
+                          commitment_bytes: Bytes48,
+                          proof_bytes: Bytes48) -> bool:
+    """
+    Given a blob and a KZG proof, verify that the blob data corresponds to the provided commitment.
+
+    Public method.
+    """
+    commitment = bytes_to_kzg_commitment(commitment_bytes)
 
-    # Create aggregated polynomial in evaluation form
-    aggregated_poly = poly_lincomb(polynomials, r_powers)
+    polynomial = blob_to_polynomial(blob)
+    evaluation_challenge = compute_challenge(blob, commitment)
 
-    # Compute commitment to aggregated polynomial
-    aggregated_poly_commitment = KZGCommitment(g1_lincomb(kzg_commitments, r_powers))
+    # Evaluate polynomial at `evaluation_challenge` (evaluation function checks for div-by-zero)
+    y = evaluate_polynomial_in_evaluation_form(polynomial, evaluation_challenge)
 
-    return aggregated_poly, aggregated_poly_commitment, evaluation_challenge
+    # Verify proof
+    proof = bytes_to_kzg_proof(proof_bytes)
+    return verify_kzg_proof_impl(commitment, evaluation_challenge, y, proof)
 ```
 
-#### `compute_aggregate_kzg_proof`
+#### `verify_blob_kzg_proof_multi`
 
 ```python
-def compute_aggregate_kzg_proof(blobs: Sequence[Blob]) -> KZGProof:
+def verify_blob_kzg_proof_multi(blobs: Sequence[Blob],
+                                commitments_bytes: Sequence[Bytes48],
+                                proofs_bytes: Sequence[Bytes48]) -> bool:
     """
-    Given a list of blobs, return the aggregated KZG proof that is used to verify them against their commitments.
+    Given a list of blobs and blob KZG proofs, verify that they correspond to the provided commitments.
+
     Public method.
     """
-    commitments = [blob_to_kzg_commitment(blob) for blob in blobs]
-    aggregated_poly, aggregated_poly_commitment, evaluation_challenge = compute_aggregated_poly_and_commitment(
-        blobs,
-        commitments
-    )
-    return compute_kzg_proof_impl(aggregated_poly, evaluation_challenge)
+    
+    commitments, evaluation_challenges, ys, proofs = [], [], [], []
```

Comment:
I think there should be a check that ensures the sequences are the same length, like `verify_kzg_proof_multi` does:

```python
assert len(blobs) == len(commitments_bytes) == len(proofs_bytes)
```

Same thing could be done for `verify_aggregate_kzg_proof_multi` too.

Reply:
Yeah done.